plutus-core/executables/debugger/Event.hs

```diff
@@ -74,38 +73,46 @@ handleDebuggerEvent driverMailbox bev@(B.VtyEvent ev) = do
             pure ()
         _ -> handleEditorEvent
 handleDebuggerEvent _driverMailbox (B.AppEvent (UpdateClientEvent cekState)) = do
-    let uplcHighlight = do
-            uplcSpan <- uplcAnn <$> case cekState of
-                Computing _ _ _ t -> Just (UPLC.termAnn t)
-                Returning _ ctx _ -> contextAnn ctx
-                _                 -> Nothing
+    let uplcHighlight :: Maybe HighlightSpan = do
+            uplcSpan <- uplcAnn <$> cekStateAnn cekState
             pure HighlightSpan
                 { _hcSLoc = B.Location (srcSpanSLine uplcSpan, srcSpanSCol uplcSpan),
                   _hcELoc = Just $ B.Location (srcSpanELine uplcSpan, srcSpanECol uplcSpan)
                 }
-    modify' $ \st -> case cekState of
-        Computing{} ->
-            st & dsUplcHighlight .~ uplcHighlight
+    let sourceHighlight :: Maybe HighlightSpan = do
+            txSpans <- txAnn <$> cekStateAnn cekState
+            -- FIXME: use some/all spans for highlighting, not just the first one
+            firstTxSpan <- S.lookupMin $ coerce txSpans
+            -- TODO: the HS_FILE supplied from the command line gets highlighted
+            -- The highlighting will not make sense or even break if the user provides
+            -- wrong HS_FILE or if the UPLC originated from multiple HS modules.
```

Comment:
`SrcSpan` contains a `FilePath`. Why aren't we just loading up the file from the `FilePath` when we highlight a `SrcSpan`?

Reply:
I discussed this before with Ziyang, and we leave it for another PR since it would not some filepath resolution checks. Currently we ignore these annotated filepaths and only use the filepath supplied at commandline.